### PR TITLE
Add more type checker performance tests.

### PR DESCRIPTION
**Pull request by rudkx**
Also move one from fast to slow based on the fact that it wasn't
representative of the original issue (which was an expression that
didn't typecheck successfully).

### CODE REVIEW

**Comment by rudkx** [comment]
@swift-ci Please smoke test OS X platform

File: test/stdlib/Character.swift [hunk 1]
```diff
@@ -310,8 +310,8 @@ CharacterTests.test(
   expectEqualMethodsForDomain(
     ascii0to126,
     ascii1to127,
-    { x in { String(x) < String($0) } } as PredicateFn,
-    { x in { String(Character(x)) < String(Character($0)) } } as PredicateFn)
```
Comment (xedin) [inline test/stdlib/Character.swift]: Nit: if `PredicateFn` is no longer useful, `typealias` could be removed as well..

**Review by xedin — APPROVED** [review]
LGTM!

File: test/stdlib/Character.swift (hunk 1 above)
Comment (rudkx) [reply test/stdlib/Character.swift]: D'oh! Yes, will push another commit to remove that.

**Pull request merged**